### DISCUSSION TITLE
DC BlueRetro Controller Receiver with Virtual Memory for Dreamcast - how to use the VMU?

### DISCUSSION THREAD

**Discussion by hearnethehunter** [darthcloud/BlueRetro#1059] [Q&A]
I recently purchased the DC BlueRetro Controller Receiver with Virtual Memory for Dreamcast. I plugged it into the Dreamcast and paired it with a bluetooth controller. However, the Dreamcast does not show a VMU is available in the port where the DC BlueRetro Controller is plugged in. Is there something that I have to do in order to set up the VMU? Sorry if it's a silly question but I'm not very technical. The product is very new so I was unable to find a similar question. Thanks

**Comment by L0newolf69** [comment]
It has memory for power up on what joystick you use . VMU was never presented as such . I thought same thing at first .

**Reply by hearnethehunter** [reply]
That's really strange as the sales information on the website says "Not only that, but it also adds a virtual memory card for your saves, that you can download and backup through its web interface!" - I certainly read that as being being able to save game progress. The instructions also say to use the adapter in port 1 if playing a one-player game.

**Reply by cecoates** [reply]
The VMU isn't "enabled" by default so to speak. You'll want to connect to the BlueRetro via Bluetooth and open the Advanced config:

https://blueretro.io/advance.html

After you connect scroll down to "Output config" and change "Accessories" to either "Memory" or "Both" (both means memory card AND rumble).

After saving the settings, reboot the BlueRetro and plug it into your Dreamcast. Boot into the Dreamcast BIOS screen (the screen you see when there's no disc in the drive) and pair your gamepad to the controller port that you chose when editing the Output config settings (probably port 1). 

Click "File", and you should see a VMU with a big "X" on it. Select that VMU (it's the virtual VMU) and click "Delete all (memory reset)". That will format the VMU and if when you're done you see whatever VMU icon and color you chose, you're ready to go.

If you're using the universal firmware, and not a console specific device or firmware, BlueRetro uses the same space for the memory card on all systems, so formatting it for Dreamcast will erase any saves you have from another system, if you've been using the same device with other consoles:

https://github.com/darthcloud/BlueRetro/wiki/BlueRetro-BLE-Web-Config-User-Manual#7---memory-card-manager-pages

> ** FOR UNIVERSAL BLUERETRO USER ** Since all systems share the same memory on the adapter for memory card emulation, if you format this area for system X, and then use your BlueRetro on system Y and format the memory card again, you have lost your data for system X. Backup the data with this interface before switching systems.

Video with the steps:
https://youtu.be/LKvF_DWR5So?feature=shared&t=418

You may want to backup your VMU data now and then: 
https://blueretro.io/dc_vmu.html

Click "read" to save your VMU data as a bin file, that way you could possibly restore it later if you accidentally deleted it or it got corrupted.
